web/source/text/processor.ts

```diff
@@ -439,7 +464,29 @@ namespace com.keyman.text {
       
       // Should we swallow any further processing of keystroke events for this keydown-keypress sequence?
       if(LeventMatched) {
-        let transcription = outputTarget.buildTranscriptionFrom(preInputMock, keyEvent);
+        let alternates: Alternate[];
+
+        if(keyEvent.keyDistribution) {
+          let activeLayout = keyman['osk'].vkbd.layout as osk.ActiveLayout;
+          alternates = [];
+  
+          for(let pair of keyEvent.keyDistribution) {
+            let mock = Mock.from(preInputMock);
+            
+            let altKey = activeLayout.getLayer(keyEvent.kbdLayer).getKey(pair.keyId);
+            if(!altKey) {
+              console.warn("Potential fat-finger key could not be found in layer!");
```

Comment:
Where do these `console.warn()` messages go to?

Reply:
All `console.____()` output functions go to the Developer console within native KMW.  As for the mobile apps, both Android and iOS engines will pass `console.warn()` and `console.err()` to the system logs (for reference when debugging), but not `console.log()`.